win32ss/user/user32/windows/cursoricon.c

```diff
@@ -170,8 +172,21 @@ static int bitmap_info_size( const BITMAPINFO * info, WORD coloruse )
                 colors = 256;
         if (!colors && (info->bmiHeader.biBitCount <= 8))
             colors = 1 << info->bmiHeader.biBitCount;
+        /* Account for BI_BITFIELDS in BITMAPINFOHEADER(v1-v3) bmp's. The
+         * 'max' selection using biSize below will exclude v4 & v5's. */
         if (info->bmiHeader.biCompression == BI_BITFIELDS) masks = 3;
         size = max( info->bmiHeader.biSize, sizeof(BITMAPINFOHEADER) + masks * sizeof(DWORD) );
+        /* Test for BI_BITFIELDS format and either 16 or 32 bpp.
+         * If so, account for the 3 DWORD masks (RGB Order).
+         * BITMAPCOREHEADER tested above has no 16 or 32 bpp types.
+         * See table "All of the possible pixel formats in a DIB"
+         * at https://en.wikipedia.org/wiki/BMP_file_format. */
+        if (info->bmiHeader.biSize >= sizeof(BITMAPV4HEADER) &&
+            info->bmiHeader.biCompression == BI_BITFIELDS &&
+            (info->bmiHeader.biBitCount == 16 || info->bmiHeader.biBitCount == 32))
+        {
+            size += 3 * sizeof(DWORD);  // BI_BITFIELDS
+        }
```

Comment:
Question: Is it possible to surround all the actual code changes with a compile-time-switch that makes the changes only effective to MSVC and prevents any negative performance impact for GCC?
Here and at the other location where actual code-statements were added.

Reply:
Thank you for your explanation. I guess it is good for now as it is. You can consider may question as resolved, but let's keep this thread expanded for future eyes.

Reply:
@JoachimHenze, Yes, an alternative workaround would be to replace the bitmap with one without the BI_BITFIELD compression and this should be enough. But reducing the Bits per Plane to be closer to what Windows uses would be good as well.